Source/mhd/mhd_plm.cpp

```diff
@@ -186,13 +162,13 @@ Castro::plm(const Box& bx,
       for (int ii = 0; ii < NEIGN; ii++) {
 
         // construct the ii-th primitive variables
-        Real W[5] = {};
+        Real W[nslp] = {};
         for (int n = 0; n < NEIGN; n++) {
-          W[0] += leig(ii,n) * Q[n][0];
-          W[1] += leig(ii,n) * Q[n][1];
-          W[2] += leig(ii,n) * Q[n][2];
-          W[3] += leig(ii,n) * Q[n][3];
-          W[4] += leig(ii,n) * Q[n][4];
+          W[im2] += leig(ii,n) * Q[n][im2];
```

Comment:
Should this construction be generalized too?

Reply:
I think that this is alreadyy as general as it can be, since this is just reconstructing the primitive variables from the characteristic variables.  There is not a load_stencil operation for that, since it is l . dq, where l is the left eigenvector.

Reply:
What I mean is that there could be an inner loop over the number of points in the stencil, so that if it were a seven point stencil you wouldn't need to add W[im3] and W[ip3] lines. Unless I am also misunderstanding what is happening inside this loop.

Reply:
I see what you are saying.  Yes, you are right.  I've updated it.  Let's see if the MHD CI test passes.